Empire/Empire.cpp

```diff
@@ -3429,6 +3429,17 @@ int Empire::TotalShipsOwned() const {
     return counter;
 }
 
+int Empire::TotalShipPartsOwned() const {
```

Comment:
You can't recalculate this every time the value is needed. It will need to be store in the empire as state info, and updated once per turn in UpdateOwnedObjectCounters